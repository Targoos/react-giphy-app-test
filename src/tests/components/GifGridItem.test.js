import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas en el componente <GifGridItem />", () => {
  const title = "Este es el titulo";
  const url = "http://localhost:4000/algo.png";

  let wrapper = shallow(<GifGridItem title={title} url={url} />);

  beforeEach(() => {
    wrapper = shallow(<GifGridItem title={title} url={url} />);
  });

  test("Debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe tener un parrafo con el titulo", () => {
    const parrafo = wrapper.find("p");

    expect(parrafo.text().trim()).toBe(title);
  });

  test("Debe tener la imagen igual a url y alt de los props", () => {
    const imagen = wrapper.find("img");

    expect(imagen.props().src).toBe(url);
    expect(imagen.props().alt).toBe(title);
  });

  test("Debe tener animate__fadeIn", () => {
    const imagen = wrapper.find("div");

    expect(imagen.hasClass("animate__fadeIn")).toBe(true);
  });
});
