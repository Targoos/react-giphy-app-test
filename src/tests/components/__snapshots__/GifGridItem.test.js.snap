// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Este es el titulo"
    src="http://localhost:4000/algo.png"
  />
  <p>
     
    Este es el titulo
     
  </p>
</div>
`;
